src/Services/SettingsBase.cs

```diff
@@ -247,6 +267,25 @@ private static async Task<string> ReadFileAsStringAsync(IFileData file, Cancella
         /// Raised when an exception is thrown during <see cref="SaveAsync(CancellationToken?)"/>.
         /// </summary>
         public event EventHandler<SettingPersistFailedEventArgs>? SaveFailed;
+
+        /// <summary>
+        /// A wrapper that holds settings data.
+        /// </summary>
+        /// <param name="Type">The type of setting.</param>
+        /// <param name="Data">The value of the setting.</param>
+        /// <param name="IsDirty">Determines whether the <paramref name="Data"/> was modified or not.</param>
+        public record SettingValue(Type Type, object Data, bool IsDirty = true)
```

Comment:
You've added a Type property but aren't using it anywhere. Let's remove it if not needed.

Reply:
It is used twice in the code (ln141, ln162)

Reply:
I'm not seeing this used on 141
![image](https://user-images.githubusercontent.com/9384894/184424489-7fee1a7a-2727-43b6-a785-832a2db9618c.png)

I do it used on 146, just missed it! All good here 👍